yt_dlp/extractor/plex.py

```diff
@@ -0,0 +1,471 @@
+import urllib.parse
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    determine_ext,
+    float_or_none,
+    int_or_none,
+    parse_age_limit,
+    traverse_obj,
+)
+
+
+class PlexWatchBaseIE(InfoExtractor):
+    _NETRC_MACHINE = 'plex'
+    # provider url is in https://plex.tv/media/providers?X-Plex-Token=<plex_token>
+    # nb: need Accept: application/json, otherwise return xml
+    _CDN_ENDPOINT = {
+        'vod': 'https://vod.provider.plex.tv',
+        'live': 'https://epg.provider.plex.tv',
+        # from api ( identifier : baseUrl)
+        'tv.plex.provider.epg': 'https://epg.provider.plex.tv',
+        'tv.plex.provider.vod': 'https://vod.provider.plex.tv',
+        'tv.plex.provider.music': 'https://music.provider.plex.tv',
+        # not used yet, but will be supported
+        'tv.plex.provider.discover': 'https://discover.provider.plex.tv',
+        'tv.plex.provider.metadata': 'https://metadata.provider.plex.tv',
+
+    }
+
+    _TOKEN = None
+    _CLIENT_IDENTIFIER = None
+
+    def _initialize_pre_login(self):
+        # TO DO: find better way to get cookie
+        # request to random page in plex.tv to get clientIdentifier in cookie
+        if self._TOKEN is None:
+            client_id = self._request_webpage(  # noqa: F841
+                'https://watch.plex.tv/', 'client_id', note='Downloading html page to get clientIdentifier')
+            self.write_debug('trying to get clientIdentifier')
+
+            # get cookie from cookiejar
+            cookie_ = {cookie.name: cookie.value for cookie in self.cookiejar}
+            self._CLIENT_IDENTIFIER = cookie_.get('clientIdentifier')
```

Comment:
```suggestion
            self._CLIENT_IDENTIFIER = self._get_cookies('https://watch.plex.tv/').get('clientIdentifier')
```

Reply:
This one give None at cookies. Here's the log:
```
(python_proj2) ~>python -m yt_dlp -v test:plexwatchmovie --test
[debug] Command-line config: ['-v', 'test:plexwatchmovie', '--test']
[debug] Portable config "~\yt-dlp.conf": ['--netrc-location', '~yt-dlp_netrc']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.06.29 [9d339c41e] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: dc7449f80
[debug] Python 3.10.4 (CPython 64bit) - Windows-10-10.0.19044-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1-full_build-www.gyan.dev (setts), ffprobe 5.0.1-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.14.1, brotli-1.0.9, certifi-2022.05.18.1, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] [TestURL] Extracting URL: test:plexwatchmovie
[TestURL] Test URL: https://watch.plex.tv/movie/bowery-at-midnight
[debug] [PlexWatchMovie] using non-login method (login as anonymous)
ERROR: 'NoneType' object has no attribute 'encode'
Traceback (most recent call last):
  File "~\yt_dlp\YoutubeDL.py", line 1431, in wrapper
    return func(self, *args, **kwargs)
  File "~\yt_dlp\YoutubeDL.py", line 1501, in __extract_info
    ie_result = ie.extract(url)
  File "~\yt_dlp\extractor\common.py", line 638, in extract
    self.initialize()
  File "~\yt_dlp\extractor\common.py", line 547, in initialize
    self._real_initialize()
  File "~\yt_dlp\extractor\plex.py", line 70, in _real_initialize
    'X-Plex-Client-Identifier': self._CLIENT_IDENTIFIER.encode()
AttributeError: 'NoneType' object has no attribute 'encode'
```